src/main/kotlin/org/rust/ide/annotator/fixes/CreateStructFieldFromConstructorFix.kt

```diff
@@ -0,0 +1,82 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.ide.annotator.fixes
+
+import com.intellij.codeInspection.LocalQuickFixAndIntentionActionOnPsiElement
+import com.intellij.openapi.editor.Editor
+import com.intellij.openapi.project.Project
+import com.intellij.psi.PsiElement
+import com.intellij.psi.PsiFile
+import com.intellij.psi.util.PsiTreeUtil
+import org.rust.lang.core.psi.*
+import org.rust.lang.core.psi.ext.parentStructLiteral
+
+class CreateStructFieldFromConstructorFix(
+    structLiteralField: RsStructLiteralField
+) : LocalQuickFixAndIntentionActionOnPsiElement(structLiteralField) {
+    override fun getText() = "Create field"
+
+    override fun getFamilyName(): String = text
+
+    override fun invoke(
+        project: Project,
+        file: PsiFile,
+        editor: Editor?,
+        startElement: PsiElement,
+        endElement: PsiElement
+    ) {
+        val field = startElement as RsStructLiteralField
+        val (fieldName, fieldType) = field.nameAndType() ?: return
+
+        val struct = field.structItem() ?: return
+        val psiFactory = RsPsiFactory(project)
+        val structBlockFields = struct.blockFields
+        if (structBlockFields != null) {
+            val rBrace = structBlockFields.rbrace ?: return
+            ensureTrailingComma(structBlockFields.namedFieldDeclList)
+            structBlockFields.addBefore(psiFactory.createBlockFields(fieldName, fieldType).children.first(), rBrace)
+        } else {
+            val identifier = struct.identifier ?: return
+            struct.addAfter(psiFactory.createBlockFields(fieldName, fieldType), identifier)
+            struct.semicolon?.delete()
+        }
+    }
+
+    private fun RsPsiFactory.createBlockFields(name: String, type: RsTypeReference): RsBlockFields {
+        return createBlockFields(listOf(RsPsiFactory.BlockField(false, name, type)))
+    }
+
+    companion object {
+        fun tryCreate(field: RsStructLiteralField): CreateStructFieldFromConstructorFix? {
+            if (field.nameAndType() == null) return null
+            val struct = field.structItem() ?: return null
+            val structBlockFields = struct.blockFields
+            if (structBlockFields != null) {
+                if (structBlockFields.rbrace == null) return null
+            } else {
+                if (struct.identifier == null) return null
+            }
+            return CreateStructFieldFromConstructorFix(field)
+        }
+
+        private fun RsStructLiteralField.structItem(): RsStructItem? {
+            return parentStructLiteral.path.reference.resolve() as? RsStructItem
+        }
+
+        private fun RsStructLiteralField.nameAndType(): Pair<String, RsTypeReference>? {
```

Comment:
I think we should use `Ty` instead of `RsTypeReference`. This will help us support a more complex case, like this:

```rust
fn foo((x, y): (i32, i32)) {
    S { x }
}
```
You can simply get the field's type:

```rust
private fun RsStructLiteralField.inferFieldTy(): Ty {
    val type = if (colon == null) resolveToBinding()?.type else expr?.type
    return type ?: TyUnknown
}
```

And then check if we can use it in the context of struct item:

```rust
fun canUse(ty: Ty): Boolean {
    val result = ty.visitWith(object : TypeVisitor {
        override fun visitTy(ty: Ty): Boolean = when (ty) {
            is TyUnknown, is TyTypeParameter, is TyAnon, is TyInfer, is TyProjection -> true
            else -> ty.superVisitWith(this)
        }

        override fun visitRegion(region: Region): Boolean = region !is ReStatic
    })
    return !result
}
```

This function returns `false` in these cases:

```rust
struct St {}
trait Tr {}

fn foo<T>(x: T, t: impl Tr, z: &str) {
    St { x, y, z }
}
```

p.s. You can use `insertionSafeTextWithLifetimes` in `createBlockFields` to render `Ty`.